### ISSUE TITLE
Add a way to skip CI with github integration when commits contain `[ci skip]`

### ISSUE THREAD

**Issue by Eijebong** [taskcluster/taskcluster#5311]
Most CI systems have that, it'd be nice to have for the github integration. I could implement that in my decision task of course but that would still require workers to run that and would change the CI status.

**Comment by lotas** [comment]
Checking [github docs](https://docs.github.com/en/actions/managing-workflow-runs/skipping-workflow-runs), they support for the workflows/actions:
* [skip ci]
* [ci skip]
* [no ci]
* [skip actions]
* [actions skip]

[Circle CI](https://circleci.com/docs/skip-build) supports those:
* [skip ci]
* [ci skip]

[Gitlab](https://docs.gitlab.com/ee/ci/pipelines/#skip-a-pipeline) supports (although we don't yet support it):
* [skip ci]
* [ci skip]

So I guess TC should support this lowest denominator of `[ci skip]`/`[skip ci]` phrases in commit messages

**Comment by lotas** [comment]
There are few things worth discussing:

1. should it work if you push several commits at once?
* `commit1: do some work`
* `commit2: do extra changes`
* `commit3: [skip ci] this is scratch..`

Should we run the tasks? Should we just check if one of the commits have it? Should all commits have it? Should just last commit have it?

2. Should we add the same for the `pull_request` events, so when you open a PR that has `[skip ci]` in title, should we tell Taskcluster to also ignore it?
I feel like it would be then required to track the title change all the time or we just create a fake check suite with single check run that will fail with `[x] ci was skipped` so this way you can never merge it before you push something meaningful that passes tests.

**Comment by petemoore** [comment]
I think this is a great idea.

> There are few things worth discussing:
> 
> 1. should it work if you push several commits at once?
> 
> * `commit1: do some work`
> * `commit2: do extra changes`
> * `commit3: [skip ci] this is scratch..`
> 
> Should we run the tasks? Should we just check if one of the commits have it? Should all commits have it? Should just last commit have it?

We can look at other CI systems for prior art regarding the best approach here. Also [try syntax](https://firefox-source-docs.mozilla.org/taskcluster/try.html#try-option-syntax).

When a push contains multiple new commits, we run a single taskgraph, so task graphs are really per-push rather than per-commit. The task runs for the commit that was pushed, so I feel this should be the one and only commit message that should be parsed. So to summarise:

* Only commits that are _pushed_ trigger tasks. Commits that aren't directly pushed don't trigger tasks. This is existing behaviour
* The commit message of the pushed commit determines behaviour (e.g. to skip or not to skip)
* Let's be mindful of future enhancements to the commit syntax parsing; like try syntax, we might wish to allow finer tuning in future, for example, running a subset of tasks, ignoring specific platforms, only running certain types of tests, running additional tests (such as performance tests or benchmarks), ....



> 
> 2. Should we add the same for the `pull_request` events, so when you open a PR that has `[skip ci]` in title, should we tell Taskcluster to also ignore it?
>    I feel like it would be then required to track the title change all the time or we just create a fake check suite with single check run that will fail with `[x] ci was skipped` so this way you can never merge it before you push something meaningful that passes tests.

I think the commit message is the better place to do this, as it can be controlled per-commit. If we one day want a pull request integration, I think it would be better to have a bot that you can message with a comment of the PR, so that the titles won't be polluted.
